geo-traits/src/line_string.rs

```diff
@@ -0,0 +1,54 @@
+use geo_types::{Coord, CoordNum, LineString};
+
+use super::CoordTrait;
+use std::iter::Cloned;
+use std::slice::Iter;
+
+pub trait LineStringTrait<'a>: Send + Sync {
+    type ItemType: 'a + CoordTrait;
+    type Iter: Iterator<Item = Self::ItemType>;
```

Comment:
```suggestion
    type Iter: ExactSizeIterator<Item = Self::ItemType>;
```

Since we already require the exact count, I wonder if we should just do `ExactSizeIterator` here

Reply:
In fact if we did this, we probably wouldn't need a `num_coords` method

Reply:
I updated to use `ExactSizeIterator` but didn't yet remove `num_coords`